lib/internal/mime.js

```diff
@@ -300,15 +416,21 @@ class MIMEType {
   #type;
   #subtype;
   #parameters;
+  #essence;
   constructor(string) {
-    string = `${string}`;
-    const data = parseTypeAndSubtype(string);
-    this.#type = data.type;
-    this.#subtype = data.subtype;
+    // #1 Remove any leading and trailing HTTP whitespace from input
+    string = StringPrototypeTrim(`${string}`);
```

Comment:
```suggestion
    typeof string !== 'string' && (string = `${string}`)
    string = StringPrototypeTrim(string);
```

Reply:
`%String.prototype.trim%` always call `ToString` per spec, it shouldn't be necessary to to convert it first.
```suggestion
    string = StringPrototypeTrim(string);
```